### PR TITLE
Add a metadata2gha script

### PR DESCRIPTION
**Pull request by ekohl**
This script generates setfiles on stdout in a format usable by Github actions.

### CODE REVIEW

**Review by bastelfreak — APPROVED** [review]

**Comment by ekohl** [comment]
Going to try and add Puppet major versions as well. Might even take a stab at the excludes.

File: bin/metadata2gha-beaker [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  exit 2
+end
+
+setfiles = []
+metadata.beaker_setfiles(use_fqdn: options[:'use-fqdn'], pidfile_workaround: options[:'pidfile-workaround']) do |setfile, name|
+  setfiles << {name: name, value: setfile}
+end
+
+puts "::set-output name=setfiles::#{setfiles.to_json}"
+puts "::set-output name=puppet_major_versions::#{metadata.puppet_major_versions}"
```
Comment (ekohl) [inline bin/metadata2gha-beaker:45]: I am considering also making the major versions have a `name` and `value`. They should also be sorted descending so the newest versions are tested first.

**Comment by ekohl** [comment]
Going to generalize it further.

File: lib/puppet_metadata/github_actions.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        {
+          puppet: puppet,
+          ruby: ruby,
+        }
+      end
+    end
+
+    def puppet_ruby_version(puppet_version)
+      case puppet_version
+      when 4
```
Comment (bastelfreak) [inline lib/puppet_metadata/github_actions.rb:54]: I don't mind having puppet 4 in it, but I also think we could drop it

File: lib/puppet_metadata/github_actions.rb [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        '2.1'
+      when 5
+        '2.4'
+      when 6
+        '2.5'
+      when 7
+        '2.7'
+      end
```
Comment (bastelfreak) [inline lib/puppet_metadata/github_actions.rb:62]: should we add a default statement here that points to `2.7`?

File: lib/puppet_metadata/github_actions.rb (hunk 2 above)
Comment (ekohl) [reply lib/puppet_metadata/github_actions.rb:54]: I thought it was nice to have for completeness, even though we shouldn't use it anymore.

File: lib/puppet_metadata/github_actions.rb (hunk 3 above)
Comment (ekohl) [reply lib/puppet_metadata/github_actions.rb:62]: I don't think so. Right now it should skip unknown versions (untested actually, will add a test for that) which means that if you still have Puppet 3 listed, it shouldn't attempt to test it with Ruby 2.7.

Comment (ekohl) [reply lib/puppet_metadata/github_actions.rb:64]: Updated with an added `compact` and tests.

**Review by bastelfreak — APPROVED** [review]

**Pull request merged**